code/game/objects/items/weapons/tanks/jetpack.dm

```diff
@@ -65,6 +66,12 @@
 		src.ion_trail.stop()
 		return 0
 
+	if((num < 5))
+		if(src.air_contents.total_moles <num && warned != 1)
+			warned = 1
+			playsound(user, 'sound/effects/alert.ogg', 50, 1)
+			user << "<span class='danger'>The meter on \the [src] indicates you are almost out of gas and beeps loudly!</span>"

```

Comment:
The issue with this is. It will only play the sound once. Per jetpack. Ever. Because warned is never nulled again. And also, warnednogas is never used.

Either use a spawn() proc in here to clear the warning after like, 1 minute. Or do something else.
```
	if (src.air_contents.total_moles < 5 && !warned)
		warned = 1
		playsound(user, 'sound/effects/alert.ogg', 50, 1)
		user << "<span class='danger'>The meter on \the [src] indicates you are almost out of gas and beeps loudly!</span>"
		spawn(600)
			warned = 0
```

Also, make sure to play test. The if checks, specially with the way num is set up, look iffy to me. They may be triggerable at invalid times.